lib/tortoise311/connection/inflight.ex

```diff
@@ -360,18 +360,15 @@ defmodule Tortoise311.Connection.Inflight do
 
   # Assign a random identifier to the tracked package; this will make
   # sure we pick a random number that is not in use
+  # The return value is guaranteed to be >= 1 and <= 65535;
+  # it is never 0 so that doesn't need to be checked.
   defp assign_identifier(%{identifier: nil} = package, pending) do
-    case :crypto.strong_rand_bytes(2) do
-      <<0, 0>> ->
-        # an identifier cannot be zero
-        assign_identifier(package, pending)
-
-      <<identifier::integer-size(16)>> ->
-        unless Map.has_key?(pending, identifier) do
-          {:ok, %{package | identifier: identifier}}
-        else
-          assign_identifier(package, pending)
-        end
+    identifier = :rand.uniform(65535)
```

Comment:
```suggestion
    # The identifier can't be zero. The :rand.uniform call below guarantees 
    # a return value >= 1 and <= 65535, so no check for zero is needed.
    identifier = :rand.uniform(65535)
```
I probably should have just written this since it's is mostly for future-me. I basically moved your comment by the :rand.uniform function since that's what tricks me up. I always forget that it doesn't return zero.